src/utilities/filetypes/AlfalfaJSON.cpp

```diff
@@ -0,0 +1,223 @@
+#include "AlfalfaJSON.hpp"
+#include "AlfalfaJSON_Impl.hpp"
+#include "utilities/alfalfa/AlfalfaActuator.hpp"
+#include "utilities/alfalfa/AlfalfaOutputVariable.hpp"
+#include "utilities/alfalfa/AlfalfaGlobalVariable.hpp"
+#include "utilities/alfalfa/AlfalfaMeter.hpp"
+#include "utilities/alfalfa/AlfalfaConstant.hpp"
+#include "../core/PathHelpers.hpp"
+#include "utilities/idd/IddObject.hpp"
+#include "utilities/idd/IddEnums.hpp"
+
+#include <utilities/idd/OS_EnergyManagementSystem_OutputVariable_FieldEnums.hxx>
+#include <utilities/idd/OS_EnergyManagementSystem_GlobalVariable_FieldEnums.hxx>
+#include <utilities/idd/OS_EnergyManagementSystem_Actuator_FieldEnums.hxx>
+#include <utilities/idd/OS_Output_Variable_FieldEnums.hxx>
+#include <utilities/idd/OS_Output_Meter_FieldEnums.hxx>
+#include <utilities/idd/EnergyManagementSystem_OutputVariable_FieldEnums.hxx>
+#include <utilities/idd/EnergyManagementSystem_GlobalVariable_FieldEnums.hxx>
+#include <utilities/idd/EnergyManagementSystem_Actuator_FieldEnums.hxx>
+#include <utilities/idd/Output_Variable_FieldEnums.hxx>
+#include <utilities/idd/Output_Meter_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+
+#include <iostream>
+namespace openstudio {
+namespace alfalfa {
+  namespace detail {
+    AlfalfaJSON_Impl::AlfalfaJSON_Impl() {}
+
+    AlfalfaJSON_Impl::AlfalfaJSON_Impl(const std::string& s) {}
+
+    AlfalfaJSON_Impl::AlfalfaJSON_Impl(const openstudio::path& p) {}
+
+    void AlfalfaJSON_Impl::exposePoint(const AlfalfaPoint& point) {
+      m_points.push_back(point);
+    }
+
+    std::vector<AlfalfaPoint> AlfalfaJSON_Impl::getPoints() {
+      return m_points;
+    }
+
+    bool AlfalfaJSON_Impl::saveAs(const openstudio::path& path) {
+      m_JSONPath = path;
+      return save();
+    }
+
+    bool AlfalfaJSON_Impl::save() const {
+      if (makeParentFolder(m_JSONPath)) {
+        Json::Value root = toJSON();
+        std::ofstream outFile(openstudio::toSystemFilename(m_JSONPath));
+
+        if (outFile) {
+          try {
+            outFile << root;
+            outFile.close();
+            return true;
+          } catch (...) {
+            LOG(Error, "Unable to write file to path '" << toString(m_JSONPath) << "'.");
+            return false;
+          }
+        }
+      }
+      LOG(Error, "Unable to write file to path '" << toString(m_JSONPath) << "', because parent directory "
+                                                  << "could not be created.");
+
+      return false;
+    }
+
+    Json::Value AlfalfaJSON_Impl::toJSON() const {
+      Json::Value root;
+      for (AlfalfaPoint point : m_points) {
```

Comment:
Making a copy of each object, probably not on purpose. Yes, this is a pImpl class, but still, there's no point not taking a `const AlfalfaPoint& point` here